docs-source/src/architecture.rst

```diff
@@ -39,12 +65,15 @@ Grace time
 **********
 A job can take some time to check out the license from the license server after it is submitted to Slurm.
 Thus, each license has a ``grace time`` period that is used to indicate how long a booking will be retained.
-After the ``grace time`` expires, the booking is deleted. This means that the license was checked out from the
-license server and doesn't need a booking anymore.
+After the ``grace time`` expires, the booking is deleted. If the ``grace time`` expires the job is assumed to have failed
```

Comment:
The grace time indicates how long the job takes to check out the license from the license server.

Once the grace time expires, the booking is deleted because the license is already accounted for in the license server (which implies that the Slurm counter was updated with the new value from the license server).

It doesn't mean that the job has failed if the grace time expires. In case of job failure, the epilog will delete the booking.

Reply:
The grace time represents the approximate period of time that a job takes to checkout a license from the license server. After this period has passed, it's assumed that the job checked out the license from the license server and the booking is no longer required.

In the reconciliation we clean up the bookings by the grace time:
- check all jobs that are running using squeue
- get the bookings for each running job
- check if the grace time for the license booked has expired
- if it has, the booking is deleted

This creates two opportunities for a booking to be deleted: when the job finishes or when the grace time is expired, whichever comes first.